eth1data/eth1data.go

```diff
@@ -34,9 +35,14 @@ func GetEth1Transaction(hash common.Hash) (*types.Eth1TxData, error) {
 
 		data := wanted.(*types.Eth1TxData)
 		if data.BlockNumber != 0 {
-			err := db.ReaderDb.Get(&data.Epoch,
-				`select epochs.finalized, epochs.globalparticipationrate from blocks left join epochs on blocks.epoch = epochs.epoch where blocks.exec_block_number = $1 and blocks.status='1';`,
-				data.BlockNumber)
+			latestFinalizedEpoch := services.LatestFinalizedEpoch()
+			err := db.ReaderDb.Get(&data.Epoch, `
+				SELECT (epochs.epoch <= $2) AS finalized, epochs.globalparticipationrate 
+				FROM blocks 
+				LEFT JOIN epochs ON blocks.epoch = epochs.epoch 
+				WHERE blocks.exec_block_number = $1 
+				AND blocks.status='1';`,
```

Comment:
It's strange that we never use ';' at the end when in Golang, but I got used to it.
Anyway, you used here ';', which should be removed, in sake of consistency.